### DISCUSSION TITLE
Unable to pick asset from global?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5881] [Troubleshooting] [answered]
Hi

Is this an issue with runtime parser, or do I need to change the way I'm using the code below? It works with regex parser.

I've updated to `3.3.8`, and I've noticed this code is no longer working? Instead of showing one asset, all of them are output.

```
{{ placeholder }}
{{ assets.3 }}
    <img src="{{ glide:url preset='w425_h239' }}" alt="{{ alt }}">
    {{ /assets.3 }}
{{ /placeholder }}
```

I've got a global called `placeholder`, I was using the code above to cherry pick a specific asset.

```
title: Placeholder
data:
  assets:
    - products/sm24-full-width-master.jpg
    - case-studies/bushmills-landscape-masonry.jpg
    - case-studies/farr2-full-width-master.jpg
    - case-studies/fingerprint-featured.jpg
    - case-studies/hammersmith-full-width-master.jpg
```

Since updating `assets.3` is ignored,  all the images are output from `assets`?

I can output the `url` if I do this

```
{{ placeholder }}
{{ assets.3 }}
{{ /placeholder }}
```

However, I'm using  `placeholder` as a partial and I'm passing in a variable like this

```
 {{  partial src="_components/glide/placeholder/img" pick_img="3"  }}
```
Inside the partial, I was using `pick_img` like this

```
{{ placeholder }}
    {{ assets[pick_img] }}   
        {{ partial src="_components/glide/ratio/16-9" }}
    {{ /assets[pick_img] }}
{{ /placeholder }}
```

This works with the regex parser, but not the runtime parser.

**Installation**
```

 'version' => 'runtime'

Statamic 3.3.8 Pro
Laravel 9.5.1
PHP 8.0.8
simonhamp/statamic-heroicons 1.0.1
statamic/collaboration 0.4.0

```

**Comment by edalzell** [comment]
what is `assets.3` supposed to do?

**Reply by stephenmeehanuk** [reply]
Hi, 

`{{ assets.3 }}` is just a hardcoded example.

When using `regex`, this displays the third asset from my `placeholder` global. 

```
{{ placeholder }}
{{ assets.3 }}
    <img src="{{ glide:url preset='w425_h239' }}" alt="{{ alt }}">
    {{ /assets.3 }}
{{ /placeholder }}
```

Unfortunately, it doesn't work when using the new `runtime` parser. The number 3 is ignored and all the assets are shown.

As mentioned above, I'm using this as a partial. So the 3 value can be changed via a variable set on the partial, my actual code looks like this

```
{{ placeholder }}
    {{ assets[pick_img] }}   
        {{ partial src="_components/glide/ratio/16-9" }}
    {{ /assets[pick_img] }}
{{ /placeholder }}
```
There are two problems.

1. The assets value, either hardcoded `{{ assets.3 }}`or using the `pick_img` variable are ignored
2. All the assets are shown, breaking the page layout. Many images are shown, instead of one.

As a temporary fix I've added `limit="1"` to my code, this at fixes the layout issue.

```
{{ placeholder }}
    {{ assets[pick_img] limit="1" }}   
        {{ partial src="_components/glide/ratio/16-9" }}
    {{ /assets[pick_img] }}
{{ /placeholder }}
```

I use this partial as a way to provide a fallback if an image isn't picked in the CMS.

**Comment by JohnathonKoster** [comment, marked as answer]
I was able to reproduce this, and it is indeed a bug. It will be corrected with this PR: https://github.com/statamic/cms/pull/5902
